From 27490aae697879bdcf45570b3be9c39f5eb72961 Mon Sep 17 00:00:00 2001
From: Evgeny Vereshchagin <evvers@ya.ru>
Date: Tue, 3 Apr 2018 16:04:22 +0300
Subject: [PATCH] core: skip the removal of cgroups in the TEST_RUN_MINIMAL
 mode (#8622)

When `systemd` is run in the TEST_RUN_MINIMAL mode, it doesn't really
set up cgroups, so it shouldn't try to remove anything.

Closes https://github.com/systemd/systemd/issues/8474.

(cherry picked from commit f6c63f6fc90040f0017a7cc37f3a05d5b86226d7)
---
 src/core/cgroup.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/core/cgroup.c b/src/core/cgroup.c
index 65ed86580f..a923ab8a76 100644
--- a/src/core/cgroup.c
+++ b/src/core/cgroup.c
@@ -2306,7 +2306,7 @@ void manager_shutdown_cgroup(Manager *m, bool delete) {
 
         /* We can't really delete the group, since we are in it. But
          * let's trim it. */
-        if (delete && m->cgroup_root)
+        if (delete && m->cgroup_root && m->test_run_flags != MANAGER_TEST_RUN_MINIMAL)
                 (void) cg_trim(SYSTEMD_CGROUP_CONTROLLER, m->cgroup_root, false);
 
         m->cgroup_empty_event_source = sd_event_source_unref(m->cgroup_empty_event_source);
-- 
2.17.0

